Source/WebCore/platform/text/cocoa/LocalizedDateCache.mm

```diff
@@ -136,6 +131,31 @@ static void _localeChanged(CFNotificationCenterRef, void*, CFStringRef, const vo
 // We assume no formatter option shows that information.
 float LocalizedDateCache::calculateMaximumWidth(DateComponentsType type, const MeasureTextClient& measurer)
 {
+#if ENABLE(INPUT_TYPE_WEEK_PICKER)
+    if (type == DateComponentsType::Week) {
+        float numLengths[10];
+        for (int i = 0; i < 10; i++)
+            numLengths[i] = measurer.measureText([NSString stringWithFormat:@"%d", i]);
+
+        int widestNum = std::distance(numLengths, std::max_element(numLengths, numLengths + 10));
+        int widestNumOneThroughFour = std::distance(numLengths, std::max_element(numLengths + 1, numLengths + 5));
+        int widestNumNonZero = std::distance(numLengths, std::max_element(numLengths + 1, numLengths + 10));
+
+        RetainPtr<NSString> weekString;
+        // W50 is an edge case here; without this check, a suboptimal choice would be made when 5 and 0 are both large and all other numbers are narrow.
+        if (numLengths[5] + numLengths[0] > numLengths[widestNumOneThroughFour] + numLengths[widestNum])
```

Comment:
What about `W51`/`W52`?

Reply:
If there was a case where W51 or W52 were the largest, W15 or W25 would be chosen which would have identical widths. W50 is special because it's mirrored version, W05, would display as "Week 5" instead of "Week 05" thus making it narrower